porting/halium_7-1/Configuring.rst

```diff
@@ -118,6 +118,218 @@ At the end of the file, append this::
 
 Now complete steps 3 and 4, taking care to remember the note on how to :ref:`Rebuild-system.img`.
 
+Bluetooth
+---------
+
+When porting to devices running older kernel versions (mainly 3.x), it is necessary to replace the kernel bluetooth stack with a newer one. This is because the newer bluetooth hardware in today's bluetooth peripheral devices often has trouble talking to the older bluetooth drivers. This can be fixed by bringing in driver code from newer Linux kernel versions. The process is called *backporting*.
+
+.. _Backports:
+
+Backporting has been greatly fascilitated by the `Linux Backports Project <https://backports.wiki.kernel.org/index.php/Main_Page>`_ which has existed for some time. This project is aimed at mainline Linux kernels and the tools (scripts) therein are not specifically tailored to Ubuntu Touch. They will consequently abort at some point during the process. However, they are the best option available, and can provide significant help all the same. The method below is based on the use of a version of these scripts which has been specially prepared by Canonical.
+
+Bluetooth backporting steps
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The steps are as follows::
+
+    1. Record bluetooth driver and settings. 
+    2. Clone/download the backports scripts.
+    3. Clone/download the kernel source from the newer kernel version you wish to backport from.
+    4. Run the script to integrate the newer sources into your kernel source tree.
+    5. Fix errors that *will* occur when the script is run.
+    6. Add a security patch.
+    7. Make necessary changes to your kernel defconfig file.
+    8. Rebuild and flash the full halium-boot.img.
+    9. Add bluetooth initializaton script, rebuild and flash system.img.
+
+.. _BT-driver:
+
+Determine driver and bluetooth settings for your device
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By the time you reach this point in the porting process, you will have completed building halium-boot (probably a number of times). It is wise to note all kernel defconfig settings related to bluetooth before proceeding with the steps below. One of these settings designates the bluetooth driver used by your device, which you will need to know at a later stage.
+
+Although many of these settings will be collected in one place in your defconfig file, some will likely be in other places as well, making it difficult to ascertain all necessary settings. The directions below may help.
```

Comment:
For me, it took me a while to catch on to this. I suppose you are talking about the second paragraph? I will remove it, if you prefer.

Reply:
Ahh. You know what, the whole thing is also only slowly sinking in for me. 

Also together with our other conversation about menuconfig. This step is all about collecting old configs and there are more or less two suggested ways: a) search manually (for what? CONFIG_BT? other stuff as well?) and note down, b) use menuconfig to find everything in one submenu. 

And it's not just 'wise' to do it's an essential step. And, oh, boy I'm slow. How about changing this whole subsection a little bit: Three paragraphs. First is intro with a bit more firm and to the point language and then one paragraph each for manual and menuconfig. Maybe we can condense menuconfig into one paragraph only, rather than five paragraphs. Smt like: 

--------------

# Record old configs

Before running the backports script you should note down all the kernel `CONFIG_xxxx` settings for bluetooth. One of these settings designates the bluetooth driver used by your device. 

You can search manually through the config file to find all bluetooth settings. Many of them will appear next to each other in one place in your config file. Some may be spread somewhere around the config file.

Alternatively you can find the settings with menuconfig::

cd out/target/product/[device]/obj/KERN_OBJ 
ARCH=arm64 make menuconfig

(If your device is armhf, use ARCH=arm instead.) Navigate to the bluetooth drivers submenu and note down all activated settings and what they do. Also note which other settings they depend on (found under Help).

--------------

maybe it could also make sense to paste what it looked like for you at the end. We're basically posting your end situation also further down. Why not post the starting point. Maybe highlight what your driver config is.



Oh and also let's align the subsections headlines. The enumeration above should really be a table of contents, ie word for word the headlines of the subsections and headlines should be short with maximum "pattern-usage". Same words for same things. How about:

1. Note down old settings
2. Download backports script
3. Download 4.2 kernel
4. Run backports script
5. Fix errors
6. Apply new settings
7. Build and flash halium-boot.img
8. Build and flash system.img

Reply:
also: https://github.com/ubports/docs.ubports.com/commit/e916ce84abb10e91149f597b4ac83f99a8eb988c#commitcomment-49267411 away to get numbering automatically

Reply:
Good points. I agree with them all. 

I can add that the manual way of finding the configs is basically guesswork unless one also goes into the kernel source and reads the Kconfig-s. That is what menuconfig does for you.

I'll try to do this asap today. It would be fabulous to get this out by the time they do the Q&A